app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/securities/EnterImporterEoriNumberController.scala

```diff
@@ -16,87 +16,93 @@
 
   import SecuritiesJourney.Checks._
 
   // Allow actions only if the MRN, RfS and ACC14 declaration are in place, and TPI04 check has been made.
   override val actionPrecondition: Option[Validate[SecuritiesJourney]] =
     Some(hasMRNAndDisplayDeclarationAndRfS)
 
-  val show: Action[AnyContent] = actionReadJourney { implicit request => journey =>
-    (if (!journey.needsDeclarantAndConsigneeEoriSubmission)
-       Redirect(routes.SelectSecuritiesController.showFirst())
-     else
-       Ok(
-         enterImporterEoriNumberPage(
-           eoriNumberForm(eoriNumberFormKey).withDefault(journey.answers.consigneeEoriNumber),
-           postAction
-         )
-       )).asFuture
-  }
+  final override val postAction: Call =
+    routes.EnterImporterEoriNumberController.submit()
+
+  final override val continueAction: Call =
+    routes.EnterDeclarantEoriNumberController.show()
+
+  final override val whenEoriInputNotRequiredAction: Call =
+    routes.SelectSecuritiesController.showFirst()
+
+  final override def modifyJourney(journey: Journey, eori: Eori): Either[String, Journey] =
+    journey.submitConsigneeEoriNumber(eori)
+
+  // val show: Action[AnyContent] = actionReadJourney { implicit request => journey =>
+  //   (if (!journey.needsDeclarantAndConsigneeEoriSubmission)
+  //      Redirect(routes.SelectSecuritiesController.showFirst())
+  //    else
+  //      Ok(
+  //        enterImporterEoriNumberPage(
+  //          eoriNumberForm(eoriNumberFormKey).withDefault(journey.answers.consigneeEoriNumber),
+  //          postAction
+  //        )
+  //      )).asFuture
+  // }
 
-  val submit: Action[AnyContent] = actionReadWriteJourney { implicit request => journey =>
-    if (!journey.needsDeclarantAndConsigneeEoriSubmission)
-      (journey, Redirect(routes.SelectSecuritiesController.showFirst())).asFuture
-    else
-      eoriNumberForm(eoriNumberFormKey)
-        .bindFromRequest()
-        .fold(
-          formWithErrors =>
-            Future.successful(
-              (
-                journey,
-                BadRequest(
-                  enterImporterEoriNumberPage(
-                    formWithErrors.fill(Eori("")),
-                    postAction
-                  )
-                )
-              )
-            ),
-          eori =>
-            Future.successful(
-              journey
-                .submitConsigneeEoriNumber(eori)
-                .fold(
-                  errors => {
-                    logger.error(s"Unable to record $eori - $errors")
-                    (journey, Redirect(baseRoutes.IneligibleController.ineligible()))
-                  },
-                  updatedJourney =>
-                    (
-                      updatedJourney,
-                      Redirect(routes.EnterDeclarantEoriNumberController.show())
-                    )
-                )
-            )
-        )
-  }
+  // val submit: Action[AnyContent] = actionReadWriteJourney { implicit request => journey =>
```

Comment:
Should we keep these commented out sections?

Reply:
of course not :-)